editor/src/clj/editor/library.clj

```diff
@@ -247,6 +248,68 @@
   (when-let [game-project-entry (locate-zip-entry zip-file "game.project")]
     (:path game-project-entry)))
 
+;; ----- Version compatibility checks -----
+
+(defn- parse-leading-int ^long [^String s]
+  (let [len (count s)]
+    (loop [i 0 n 0 found false]
+      (if (< i len)
+        (let [c (.charAt s i)]
+          (if (Character/isDigit c)
+            (recur (unchecked-inc i)
+                   (unchecked-add (unchecked-multiply n 10) (Character/digit c 10))
+                   true)
+            (if found n 0)))
+        (if found n 0)))))
+
+(defn- compare-versions
+  "Compares semantic-like versions (e.g. 1.2.3). Ignores non-digit suffixes.
+   Returns negative if v1 < v2, zero if equal, positive if v1 > v2."
+  [v1 v2]
+  (let [a (str/split (or v1 "") #"\.")
+        b (str/split (or v2 "") #"\.")
+        len (max (count a) (count b))]
+    (loop [i 0]
+      (if (< i len)
+        (let [sa (get a i "0")
+              sb (get b i "0")
+              ia (parse-leading-int sa)
+              ib (parse-leading-int sb)
+              d (- ia ib)]
+          (if (zero? d)
+            (recur (unchecked-inc i))
+            d))
+        0))))
```

Comment:
I think we should avoid duplicating code between bob/java and editor/clojure whenever possible. Here it would be quite easy to do: make `LibraryUtil.compareVersions` public, and use it instead of these functions.